From 08c5dfcffeef0ba9f2ed3c6af9310d40f920e03c Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Wed, 1 Jun 2016 16:37:28 -0400
Subject: [PATCH 0519/1722] drm/amd/dal: fix crash when adapter_srv isn't
 created

A former change that implemented VSR causes a crash on diags
because diags don't create adapter service

Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc.c b/drivers/gpu/drm/amd/dal/dc/core/dc.c
index a7b4800..4aee4c1 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc.c
@@ -433,7 +433,8 @@ struct dc *dc_create(const struct dc_init_data *init_params)
 			core_dc->res_pool.stream_enc_count);
 	core_dc->public.caps.max_links = core_dc->link_count;
 	core_dc->public.caps.max_audios = core_dc->res_pool.audio_count;
-        core_dc->public.caps.max_downscale_ratio =
+        if (core_dc->res_pool->adapter_srv)
+                core_dc->public.caps.max_downscale_ratio =
                         dal_adapter_service_get_downscale_limit(
                                         core_dc->res_pool->adapter_srv);
 
-- 
2.7.4

